packages/cli/src/packages-to-be/account.ts

```diff
@@ -81,3 +82,41 @@ export const getTotalBalance = async (
     cREAL,
   }
 }
+
+export async function getMetadataURLs(client: PublicClient, addresses: Address[]) {
+  const contract = await getAccountsContract(client)
+
+  const urlResults = await Promise.allSettled(
+    addresses.map(async (address) => {
+      const url = await contract.read.getMetadataURL([address])
+      return [address, url]
+    })
+  )
+  const filtered = urlResults
+    .filter((result) => result.status === 'fulfilled')
+    .map((result) => {
+      // we know its fulfilled but the types are not so sure
+      return (result as unknown as PromiseFulfilledResult<string[]>).value as [Address, string]
```

Comment:
oof, no way to make it nicer? would this work?:

```js
const filtered = urlResults
  .map((result) => {
    if (result.status === 'fulfilled') {
        // type should be PromiseFulfilledResult no?
    }
  })
  .filter(Boolean)

Reply:
well it should be PromiseFulfilledResult after filtering status anyway :( but ill see if what change i can make

Reply:
found an even better way id say